rca/scholarships/models.py

```diff
@@ -172,66 +172,67 @@ def get_context(self, request, *args, **kwargs):
-                            "eligable_programmes__id", flat=True
-                        )
-                    ).live()
-                ),
-                filter_by="eligable_programmes__slug__in",
-                option_value_field="slug",
-            ),
-            TabStyleFilter(
-                "Location",
-                queryset=(
-                    ScholarshipLocation.objects.filter(
-                        id__in=queryset.values_list("location_id", flat=True)
-                    )
-                ),
-                filter_by="location__slug__in",
-                option_value_field="slug",
-            ),
-        )
-
-        # Apply filters
-        for f in filters:
-            queryset = f.apply(queryset, request.GET)
-
-        # Format scholarships for template
-        results = [
-            {
-                "value": {
-                    "heading": s.title,
-                    "introduction": s.summary,
-                    "eligible_programmes": ",".join(
-                        x.title for x in s.eligable_programmes.live()
+        if "programme" in request.GET:
+            queryset = Scholarship.objects.prefetch_related(
+                "eligable_programmes", "funding_categories", "fee_statuses"
+            )
+
+            filters = (
+                ProgrammeTabStyleFilter(
+                    "Programme",
+                    queryset=(
+                        ProgrammePage.objects.filter(
+                            id__in=queryset.values_list(
+                                "eligable_programmes__id", flat=True
+                            )
+                        ).live()
                     ),
-                    "funding_categories": ",".join(
-                        x.title for x in s.funding_categories.all()
+                    filter_by="eligable_programmes__slug__in",
+                    option_value_field="slug",
+                ),
+                TabStyleFilter(
+                    "Location",
+                    queryset=(
+                        ScholarshipLocation.objects.filter(
+                            id__in=queryset.values_list("location_id", flat=True)
+                        )
                     ),
-                    "fee_statuses": ",".join(x.title for x in s.fee_statuses.all()),
-                    "value": s.value,
+                    filter_by="location__slug__in",
+                    option_value_field="slug",
+                ),
+            )
+
+            # Apply filters
+            for f in filters:
+                queryset = f.apply(queryset, request.GET)
+
+            # Format scholarships for template
+            results = [
+                {
+                    "value": {
+                        "heading": s.title,
+                        "introduction": s.summary,
+                        "eligible_programmes": ",".join(
+                            x.title for x in s.eligable_programmes.live()
```

Comment:
I _think_ `x.title` here should be the __str__ method?
I might be wrong so I'll leave it to your judgement, but I think you/we updated the __str__ to output the programme level too? Just worth checking what that renders out like.

Reply:
That was you, two years to add the degree level, but I'll take that :+1: